frontend/dockerfile/docs/reference.md

```diff
@@ -2091,10 +2091,14 @@ The following `ARG` variables are set automatically:
 - `TARGETOS` - OS component of TARGETPLATFORM
 - `TARGETARCH` - architecture component of TARGETPLATFORM
 - `TARGETVARIANT` - variant component of TARGETPLATFORM
+- `TARGETOSVERSION` - OSVersion component of TARGETPLATFORM
```

Comment:
We should not add any of the new capabilities that depend on the platform string matching until containerd has defined a string representation for such cases and fixed the currently merged `platforms.Parse()` function. See https://github.com/moby/buildkit/pull/4315/files#r1378282028

The changes in `client/llb` and `containerimage/writer` where these values are based on base image config or `platforms.DefaultSpec()` are ok. Places where it is based on string like the `--opt platform=` need to wait.